aten/src/ATen/native/Distance.cpp

```diff
@@ -189,8 +189,14 @@ Tensor _cdist_backward(const Tensor& grad, const Tensor& _x1, const Tensor& _x2,
     x2 = x2.expand(tensor2_expand_size).contiguous();
   }
 
-  TORCH_CHECK(x1.is_contiguous(), "_cdist_backward requires X1 to be contiguous");
-  TORCH_CHECK(x2.is_contiguous(), "_cdist_backward requires X2 to be contiguous");
+  // TORCH_CHECK(x1.is_contiguous(), "_cdist_backward requires X1 to be contiguous");
+  // TORCH_CHECK(x2.is_contiguous(), "_cdist_backward requires X2 to be contiguous");
+  if (!x1.is_contiguous()) {
+    x1 = x1.contiguous();
+  }
+  if (!x2.is_contiguous()) {
+    x2 = x2.contiguous();
+  }
```

Comment:
```suggestion
x1 = x1.contiguous();
x2 = x2.contiguous();
```
and delete the `contiguous()` calls before in the `if`. Note that the first thing `contiguous` does is to do this check.